transport/src/main/java/io/netty/channel/socket/nio/NioSocketChannel.java

```diff
@@ -284,7 +284,19 @@ protected void doWrite(ChannelOutboundBuffer in) throws Exception {
                     // Only one ByteBuf so use non-gathering write
                     ByteBuffer nioBuffer = nioBuffers[0];
                     for (int i = config().getWriteSpinCount() - 1; i >= 0; i --) {
-                        final int localWrittenBytes = ch.write(nioBuffer);
+                        final int localWrittenBytes;
+                        try {
+                            localWrittenBytes = ch.write(nioBuffer);
+                        } catch (IllegalArgumentException e) {
+                            // There appears to be a JDK bug that when a write operation occurs on a direct buffer,
```

Comment:
@normanmaurer - O. eclipse only has this capability when you are declaring a method/variable/calls (places where javadocs typically are) and not in code.